File: 100-Days-Series/Day 82/FindingIndices.java
public class FindingIndices {
    // Method to find all occurrences of a key in the array
    public static void getFindOccurrence(int arr[],int key,int index){
        int len=arr.length;
        // Base case: if the current index reaches the size of the array, stop recursion
        if(len==index){
            return;
        }
        // If the current element matches the key, print its index
        if(arr[index]==key){
            System.out.print(index+" ");
        }
        // Recursive call to check the next element
        getFindOccurrence(arr, key, ++index);
    }
    // Main Method
    public static void main(String [] args){
        // Let's Start Here
        int arr[]={3, 2, 4, 5, 6, 2, 7, 2, 2};
        int key=2;
        getFindOccurrence(arr, key, 0);
    }
}

File: 100-Days-Series/Day 82/FindingIndices.js

// Function to find all occurrences of a key in the array
const getfindOccurrence=(arr,key,index)=>{
    let len=arr.length;
    // Base case: if the current index reaches the size of the array, stop recursion
    if(len==index){
        return;
    }
    // If the current element matches the key, print its index
    if(arr[index]==key){
        console.log(index);
    }
    // Recursive call to check the next element
    getfindOccurrence(arr,key,++index);
}
// Let's Start Here
const arr=[3, 2, 4, 5, 6, 2, 7, 2, 2];
let key=2;
getfindOccurrence(arr,key,0);